docs/installation/server-config/security.rst

```diff
@@ -132,28 +133,130 @@ Configuring Authorization of KSQL with Kafka ACLs
+
+    The ``CLUSTER`` resource type is implicitly a literal pattern with a constant name because it refers to the entire Kafka cluster.
+
+An literal example might be ``ALLOW`` ``user Fred`` to ``READ`` the ``TOPIC`` with the ``LITERAL`` name ``users``.
+
+An prefixed example might be ``ALLOW`` ``user Fred`` to ``READ`` any ``TOPIC`` whose name is ``PREFIXED`` with ``fraud-``.
+
+The ACLs described below list a ``RESOURCE_TYPE``, resource name, ``PATTERN_TYPE``, and ``OPERATION``.
+All ACLs described are ``ALLOW`` ACLs, where the principal is the user the KSQL server has authenticated as,
+with the Apache Kafka cluster, or an appropriate group that includes the authenticated KSQL user.
+
+.. tip:: For more information about ACLs see :ref:`kafka_authorization` and for more information about interactive and
+non-interactive queries, see :ref:`restrict-ksql-interactive`.
+
+Required ACLs
+^^^^^^^^^^^^^
+
+The ACLs required are the same for both :ref:`Interactive and non-interactive (headless) KSQL clusters <restrict-ksql-interactive>`:
+
+KSQL require these ACLs:
+
+- The ``DESCRIBE_CONFIGS`` operation on the ``CLUSTER`` resource type.
+- The ``READ`` operation on any input topics.
+- The ``WRITE`` operation on any output topics.
+- The ``CREATE`` operation on any output topics that do not already exist.
+- The ``ALL`` operation on all internal ``TOPIC``s ``PREFIXED`` with ``_confluent-ksql-<ksql.service.id>``.
+- The ``ALL`` operation on all internal ``GROUP``s ``PREFIXED`` with ``_confluent-ksql-<ksql.service.id>``.
+
+KSQL prefixes all internally created topics and consumer groups with ``_confluent-ksql-<ksql.service.id>``,
+where ``<ksql.service.id>`` is the KSQL service id, as defined in the clusters configuration, and defaulting to ``default_``.
+
+How you define ACLs to allow KSQL access to the input and output topics will depend on your use case and
+whether the KSQL cluster is configured for
+:ref:`interactive <config-security-ksql-acl-interactive_post_ak_2_0>` or :ref:`non-interactive (headless) <config-security-ksql-acl-headless_post_ak_2_0>`
+
+.. _config-security-ksql-acl-headless_post_ak_2_0:
+
+Non-Interactive (headless) KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+:ref:`Non-interactive KSQL clusters <restrict-ksql-interactive>` run a known set of SQL statements, meaning the set
+of input and output topics is well defined. Add the ACLs required to allow KSQL access to these topics.
+
+.. config-security-ksql-acl-interactive_post_ak_2_0
+
+Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:ref:`Interactive KSQL clusters <restrict-ksql-interactive>` accept SQL statements from users and hence may require access
+to a wide variety of input and output topics. Add ACLs to appropriate literal and prefixed resource patterns to allow KSQL
+access to the input and output topics, as required.
+
+As the set of output topics may not be known up front, it is common to to add an ACL to allow ``ALL`` operations
```

Comment:
- Typo: "to to" -> "to"

And perhaps a better reworded text:

----

Recommendations for securing interactive KSQL clusters:

- To simplify ACL management, you should configure a default, custom topic name prefix such as ``ksql-interactive-`` for your KSQL cluster via the `<whatever>` server configuration setting. [and explain how to do this, link the config, etc.] Unless a user defines an explicit topic name in a KSQL statement, KSQL will then always prefix the name of any automatically created output topics.
- Add an ACL to allow ``ALL`` operations on ``TOPIC``s ``PREFIXED`` with the configured custom name prefix (in the example above: ``ksql-interactive-``).

----